api/v1/controllers/samples.js

```diff
@@ -197,7 +201,10 @@ module.exports = {
    *  bulk upsert request has been received.
    */
   bulkUpsertSample(req, res/* , next */) {
-    utils.noReadOnlyFieldsInReq(req, helper);
+
+    // make the name post-able
+    const readOnlyFields = helper.readOnlyFields.filter((field) => field !== 'name');
+    utils.noReadOnlyFieldsInReq(req, readOnlyFields);
     const resultObj = { reqStartTime: new Date() };
```

Comment:
This one here is just individual sample upsert, so keep it, but we should get rid of line 207. We still need to check for that stuff before doing each individual upsert, but we should not be iterating over the bulk upsert array here.